pkg/controller/instance/instance_controller.go

```diff
@@ -328,6 +331,20 @@ func createPlan(mgr manager.Manager, planName string, instance *kudov1alpha1.Ins
 		Namespace: instance.Namespace,
 	}
 
+	// first lets check if there is an active plan
+	if instance.Status.ActivePlan.Namespace != "" {
```

Comment:
why are we checking namespace and not name? If this check is supposed to make sure there is no other activeplan then `instance.Status.ActivePlan.Name` would make more sense to me

Reply:
also just checking active plan is probably not enough, is it? 🤔 what if the plan exists but is suspended or failed...

Reply:
another case where I am not sure this will be 100% correct:
- I update instance with new params
- instance controller receives update event
- it is process by UpdateFunc in `instanceEventPredicateFunc`
- it goes over existing plans, makes sure the parameters change and then it invokes `createPlan` because it realizes we need to create a plan
- the create plan with this addition actually does not re-deploy the instance even since it should have

Reply:
ah... I did mean to check `Name` and not `Namespace`...
The ActivePlan of Status is not a ptr... it is not possible to check for `nil`.   This check is only to see if a ActivePlan exist.   It DOES make the assumption that all valid ActivePlans have a Name which I think is a reasonable expectation.   I believe all the cases you raise are covered.   This check is only if a "createPlan" is called.

Reply:
yeah I think the case of upgrades is still not covered and probably some others as well. The failing tests are proving that as well, see e.g. the `operator-test` failure. I think to do upgrades with this patch, we will need to do some changes in PlanExecution object itself, as some of them were mentioned by Gaston yesterday. So I would really recommend splitting this in two because I am afraid this part might take some time to get right...